README
  NAME

    PDF::Create - create PDF files

  DESCRIPTION

    PDF::Create allows you to create PDF documents using a large
    number of primitives, and emit the result as a PDF file or
    stream. PDF stands for Portable Document Format.

    Documents can have several pages, a table of contents, an
    information section, and many other PDF elements. More
    functionalities will be added as needs arise.

    Documents are constructed on the fly so the memory footprint is
    not tied to the size of the pages but only to their number.

    Its main advantage over the other PDF modules is that it does
    not depend on other modules and is perl only (no compiler needed).
    If you want a quick and dirty way of creating pdf's, PDF::Create
    is for you. If you need a complete Framework to create complex
    PDF stuff, you better dive into the PDF::API2 based modules.
    
    More information about this module is included in this package.

  SYNOPSIS

        use PDF::Create;

        my $pdf = new PDF::Create('filename' => 'mypdf.pdf',
                                  'Version'  => 1.2,
                                  'PageMode' => 'UseOutlines',
                                  'Author'   => 'Fabien Tassin',
                                  'Title'    => 'My title',
                             );
        my $root = $pdf->new_page('MediaBox' => [ 0, 0, 612, 792 ]);

        # Add a page which inherits its attributes from $root
        my $page = $root->new_page;

        # Prepare 2 fonts
        my $f1 = $pdf->font('Subtype'  => 'Type1',
                            'Encoding' => 'WinAnsiEncoding',
                            'BaseFont' => 'Helvetica');
        my $f2 = $pdf->font('Subtype'  => 'Type1',
                            'Encoding' => 'WinAnsiEncoding',
                            'BaseFont' => 'Helvetica-Bold');

        # Prepare a Table of Contents
        my $toc = $pdf->new_outline('Title' => 'Document',
                                    'Destination' => $page);
        $toc->new_outline('Title' => 'Section 1');
        my $s2 = $toc->new_outline('Title' => 'Section 2');
        $s2->new_outline('Title' => 'Subsection 1');

        $page->stringc($f2, 40, 306, 426, "PDF::Create");
        $page->stringc($f1, 20, 306, 396, "version $PDF::Create::VERSION");

        # Add another page
        my $page2 = $root->new_page;
        $page2->line(0, 0, 612, 792);
        $page2->line(0, 792, 612, 0);

        $toc->new_outline('Title' => 'Section 3');
        $pdf->new_outline('Title' => 'Summary');

        # Add something to the first page
        $page->stringc($f1, 20, 306, 300,
                       'by Fabien Tassin <fta@oleane.net>');

        # Add the missing PDF objects and a the footer then close the file
        $pdf->close;


  INSTALLATION

    Quick answer:

        perl -MCPAN -e 'install PDF::Create'

    Long answer:

    To install manually, cd to the directory containing the unpacked
    distribution and do one of the following:

    a.  Create a makefile by running Makefile.PL using the perl
        program into whose library you want to install. Then, run
        make three times:

            perl Makefile.PL
            make
            make test
            make install

    b.  To install into a private library, for example your home
        directory:

            perl Makefile.PL INSTALLSITELIB=$HOME/lib INSTALLMAN3DIR=$HOME/man
            make
            make test
            make pure_install

  AUTHORS

    - Fabien Tassin

    Original Author of PDF::Create

    - Markus Baertschi, markus@markus.org

    I have taken over maintenence of PDF::Create as Fabien has disappeared
    and did no longer maintain it in the last years.
    The last version of PDF::Create from Fabien is 0.06. All newer versions
    have been modified by me.

    I maintain PDF::Create in git. You can access the repository directly
    at http://github.com/markusb/pdf-create.




CHANGES
1.10 2014.08.22
    Fix test that used to use CGI.pm

1.09 2014.08.21
    Eliminate the need for CGI.pm in the test RT #98222
    Some POD updates.
    Eliminate memory leak RT #93412 thanks to Chris Jack.

1.08 2013.12.11

    Eliminate pipe from CGI test script t/09
    Filehandle => FileHandle typo fixed #91312

1.07 2013.12.10

    New maintainer SZABGAB
    Tried to reconstruct entries in the CHANGES file 
    Some more test case
    Code clean-up (remove indirect calls)
    Remove Build.PL
    use warnings;
    Officially require perl 5.006.
    Move example file to the eg/ directory

1.06 2010.07.06

1.05 2010.02.26

1.04 2009.07.13

1.03 200 2009.03.24

1.02 208.07.10

1.01 2008.06.03

1.00 2008.05.31

0.10 2008.05.31


0.09 2007.09.30

    - Markus Baertschi, markus@markus.org
    - Added stringml for multi-line strings
    - Changed get_page_size to case-independent

0.08 2007.09.03

    - Markus Baertschi, markus@markus.org
    - Get back in sync with CPAN/PAUSE
    - Fixed check for open file in Create.pm
    - Touched up the README

0.07

    - Markus Baertschi, markus@markus.org
    - Added image samples to sample.pl
    - Touched up the README
    - Added check for open file in JPEG/GIFImages

0.06

    - Fabien Tassin
    - Added support for JPEG and GIF images
    - Markus Baertschi, markus@markus.org
    - Packaged for release on CPAN

0.01

    - Fabien Tassin
    - Original Release



sample.pl
#!/usr/bin/perl -w
#
# sample PDF::Create usage
#

use strict;
use PDF::Create;

my $pdf = new PDF::Create('filename' => 'sample.pdf',
			  'Version'  => 1.2,
			  'PageMode' => 'UseOutlines',
			  'Author'   => 'John Doe',
			  'Title'    => 'Sample Document',
						);

my $root = $pdf->new_page('MediaBox' => $pdf->get_page_size('a4'));

# Prepare 2 fonts
my $f1 = $pdf->font('Subtype'  => 'Type1',
	 	    'Encoding' => 'WinAnsiEncoding',
			        'BaseFont' => 'Helvetica');
my $f2 = $pdf->font('Subtype'  => 'Type1',
		    'Encoding' => 'WinAnsiEncoding',
		    'BaseFont' => 'Helvetica-Bold');

# Prepare a Table of Content
my $toc = $pdf->new_outline('Title' => 'Sample Document');

# Add a page which inherits its attributes from $root
my $page = $root->new_page;
# Add a entry to the outline
$toc->new_outline('Title' => 'Page 1', 'Destination' => $page);

# Write some text to the page
$page->stringc($f2, 40, 306, 426, "PDF::Create");
$page->stringc($f1, 20, 306, 396, "version $PDF::Create::VERSION");
$page->stringc($f1, 20, 300, 300, 'Fabien Tassin');
$page->stringc($f1, 20, 300, 250, 'Markus Baertschi (markus@markus.org)');

# add another page
my $page2 = $root->new_page;
my $s2 = $toc->new_outline('Title' => 'Page 2', 'Destination' => $page2);
$s2->new_outline('Title' => 'GIF');
$s2->new_outline('Title' => 'JPEG');

# Draw a border around the page (A4 max is 595/842)
$page2->line(10, 10, 10, 832);
$page2->line(10, 10, 585, 10);
$page2->line(10, 832, 585, 832);
$page2->line(585, 10, 585, 832);

# Add a gif image
$page2->string($f1, 20, 50, 600, 'GIF Image:');
my $img1 = $pdf->image('pdf-logo.gif');
$page2->image('image'=>$img1, 'xscale'=>0.2,'yscale'=>0.2,'xpos'=>200,'ypos'=>600);

# Add a jpeg image
$page2->string($f1, 20, 50, 500, 'JPEG Image:');
my $img2 = $pdf->image('pdf-logo.jpg');
$page2->image('image'=>$img2, 'xscale'=>0.2,'yscale'=>0.2,'xpos'=>200,'ypos'=>500);

# Wrap up the PDF and close the file
$pdf->close;

